mmengine/runner/runner.py

```diff
@@ -56,6 +56,35 @@
 OptimWrapperType = Union[OptimWrapper, OptimWrapperDict]
 
 
+class SlicedDataset:
+
+    def __init__(self, dataset, length) -> None:
+        self.dataset = dataset
+        self.length = length
+        assert self.length <= len(self.dataset)
```

Comment:
Suggest removing this assertion since the function `len` could not apply to `IterableDataset`. While we can reserve the `__getitem__` methods since it could be useful when users need to access the `MapDataset` by index.